gallery_dl/extractor/steamgriddb.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.steamgriddb.com"""
+
+from .common import Extractor, Message
+from .. import text, exception
+
+
+BASE_PATTERN = r"(?:https?://)?(?:www\.)?steamgriddb\.com"
+LANGUAGE_CODES = (
+    "aa", "ab", "ae", "af", "ak", "am", "an", "ar", "as", "av", "ay", "az",
+    "ba", "be", "bg", "bh", "bi", "bm", "bn", "bo", "br", "bs", "ca", "ce",
+    "ch", "co", "cr", "cs", "cu", "cv", "cy", "da", "de", "dv", "dz", "ee",
+    "el", "en", "eo", "es", "et", "eu", "fa", "ff", "fi", "fj", "fo", "fr",
+    "fy", "ga", "gd", "gl", "gn", "gu", "gv", "ha", "he", "hi", "ho", "hr",
+    "ht", "hu", "hy", "hz", "ia", "id", "ie", "ig", "ii", "ik", "io", "is",
+    "it", "iu", "ja", "jv", "ka", "kg", "ki", "kj", "kk", "kl", "km", "kn",
+    "ko", "kr", "ks", "ku", "kv", "kw", "ky", "la", "lb", "lg", "li", "ln",
+    "lo", "lt", "lu", "lv", "mg", "mh", "mi", "mk", "ml", "mn", "mr", "ms",
+    "mt", "my", "na", "nb", "nd", "ne", "ng", "nl", "nn", "no", "nr", "nv",
+    "ny", "oc", "oj", "om", "or", "os", "pa", "pi", "pl", "ps", "pt", "qu",
+    "rm", "rn", "ro", "ru", "rw", "sa", "sc", "sd", "se", "sg", "si", "sk",
+    "sl", "sm", "sn", "so", "sq", "sr", "ss", "st", "su", "sv", "sw", "ta",
+    "te", "tg", "th", "ti", "tk", "tl", "tn", "to", "tr", "ts", "tt", "tw",
+    "ty", "ug", "uk", "ur", "uz", "ve", "vi", "vo", "wa", "wo", "xh", "yi",
+    "yo", "za", "zh", "zu",
+)
+FILE_EXT_TO_MIME = {
+    "png": "image/png",
+    "jpeg": "image/jpeg",
+    "jpg": "image/jpeg",
+    "webp": "image/webp",
+    "ico": "image/vnd.microsoft.icon",
+    "all": "all",
+}
+
+
+class SteamgriddbExtractor(Extractor):
+    """Base class for SteamGridDB"""
+    category = "steamgriddb"
+    directory_fmt = ("{category}", "{subcategory}", "{game[id]}")
+    filename_fmt = "{game[id]}_{id}_{num:>02}.{extension}"
+    archive_fmt = "{filename}"
+    root = "https://www.steamgriddb.com"
+
+    def _init(self):
+        self.cookies_update({
+            "userprefs": "%7B%22adult%22%3Afalse%7D",
```

Comment:
`{"adult":false}` by default?
Extractors usually extract all media with default settings.

Reply:
Setting it to false actually disables filtering, rather than enabling.

![](https://github.com/mikf/gallery-dl/assets/45762106/cc620d1d-26a0-45f3-8ebb-d52722e31cd5)
![](https://github.com/mikf/gallery-dl/assets/45762106/d49a4f33-b987-49cf-b797-1446da3ded7b)